radix-engine/src/model/auth_converter.rs

```diff
@@ -0,0 +1,231 @@
+use crate::model::method_authorization::{
+    HardAuthRule, HardCount, HardDecimal, HardProofRule, HardProofRuleResourceList, HardResourceOrNonFungible,
+};
+use crate::model::MethodAuthorization;
+use sbor::any::Value;
+use sbor::*;
+use scrypto::engine::types::*;
+use scrypto::prelude::{AuthRuleNode, MethodAuth, SoftResource};
+use scrypto::resource::{
+    NonFungibleAddress, ProofRule, SoftCount, SoftDecimal, SoftResourceOrNonFungible, SoftResourceOrNonFungibleList,
+};
+use scrypto::rust::vec::Vec;
+use scrypto::types::ScryptoType;
+
+fn soft_to_hard_decimal(schema: &Type, soft_decimal: &SoftDecimal, dom: &Value) -> HardDecimal {
+    match soft_decimal {
+        SoftDecimal::Static(amount) => HardDecimal::Amount(amount.clone()),
+        SoftDecimal::Dynamic(schema_path) => {
+            let sbor_path = schema_path.to_sbor_path(schema);
+            if let None = sbor_path {
+                return HardDecimal::SoftDecimalNotFound;
+            }
+            match sbor_path.unwrap().get_from_value(dom) {
```

Comment:
minor: `if let Some(dom)` can avoid a unwrap.

Reply:
Is rust smart enough to know that the negation of that has already been handled though?

Reply:
It's not